### PR TITLE
First pass at dask.imperative docs

### PR DESCRIPTION
**Pull request by jcrist**
Adds docs for dask imperative.

### CODE REVIEW

File: docs/source/imperative.rst [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+This builds the same graph as seen before, but using normal python syntax. In
+fact, the only difference between python code that would do this in serial, and
+the parallel version with dask is the ``do`` decorators on the functions, and
+the call to ``compute`` at the end.
+
+
+How it works
+------------
+
+The imperative consists of two functions:
```
Comment (mrocklin) [inline docs/source/imperative.rst]: `The imperative` seems odd

File: docs/source/imperative.rst [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+Operations that aren't supported include:
+
+- Mutating operators (``a += 1``)
+- Mutating magics such as ``__setitem__``/``__setattr__`` (``a[0] = 1``, ``a.foo = 1``)
+- Iteration. (``for i in a: ...``)
+- Use as a predicate (``if a: ...``)
+
+The last two in particular mean that ``Value`` objects cannot be used for
+control flow, meaning that no ``Value`` can appear in a loop or if statement.
+Even with this limitation, many workflows can be easily parallelized.
```
Comment (mrocklin) [inline docs/source/imperative.rst]: So far this seems really clear and direct.

File: docs/source/imperative.rst (hunk 1 above)
Comment (jcrist) [inline docs/source/imperative.rst]: woops, word missing there.

**Comment by mrocklin** [comment]
On a first pass through I actually have no comments at all.  I'm not sure that's ever happened... :)

+1

**Comment by jcrist** [comment]
That's nice to hear :). Made a touchup, good to merge?

**Comment by mrocklin** [comment]
+1 from me

**Pull request merged**